awspub/api.py

```diff
@@ -0,0 +1,165 @@
+import logging
+import pathlib
+from typing import Dict, Iterator, List, Optional, Tuple
+
+from awspub.context import Context
+from awspub.image import Image, _ImageInfo
+from awspub.s3 import S3
+
+logger = logging.getLogger(__name__)
+
+
+def _images_grouped(
+    images: List[Tuple[str, Image, Dict[str, _ImageInfo]]], group: Optional[str]
+) -> Tuple[Dict[str, Dict[str, str]], Dict[str, Dict[str, Dict[str, str]]]]:
```

Comment:
You could define your custom types here. Like:

```python3
type ImageMetadata = Dict[str, str]
```

Reply:
`type` is >= python 3.12  (see https://docs.python.org/3/reference/simple_stmts.html#type ). but we want to support 3.10 (from jammy) for now.